compiler/qsc_frontend/src/typeck/tests.rs

```diff
@@ -2190,6 +2190,309 @@ fn newtype_cons_wrong_input() {
+            }
+        "},
+        "",
+        &expect![[r#"
+            #19 76-78 "()" : Unit
+            #23 86-177 "{\n        let pair = new Pair { First = 5, Second = 6 };\n        new Pair { ...pair }\n    }" : UDT<"Pair": Item 1>
+            #25 100-104 "pair" : UDT<"Pair": Item 1>
+            #27 107-141 "new Pair { First = 5, Second = 6 }" : UDT<"Pair": Item 1>
+            #32 126-127 "5" : Int
+            #35 138-139 "6" : Int
+            #37 151-171 "new Pair { ...pair }" : UDT<"Pair": Item 1>
+            #40 165-169 "pair" : UDT<"Pair": Item 1>
+        "#]],
+    );
+}
+
+#[test]
+fn struct_copy_cons_with_fields() {
+    check(
+        indoc! {"
+            namespace A {
+                struct Pair { First : Int, Second : Int }
+                function Foo() : Pair {
+                    let pair = new Pair { First = 5, Second = 6 };
+                    new Pair { ...pair, First = 7 }
+                }
+            }
+        "},
+        "",
+        &expect![[r#"
+            #19 76-78 "()" : Unit
+            #23 86-188 "{\n        let pair = new Pair { First = 5, Second = 6 };\n        new Pair { ...pair, First = 7 }\n    }" : UDT<"Pair": Item 1>
+            #25 100-104 "pair" : UDT<"Pair": Item 1>
+            #27 107-141 "new Pair { First = 5, Second = 6 }" : UDT<"Pair": Item 1>
+            #32 126-127 "5" : Int
+            #35 138-139 "6" : Int
+            #37 151-182 "new Pair { ...pair, First = 7 }" : UDT<"Pair": Item 1>
+            #40 165-169 "pair" : UDT<"Pair": Item 1>
+            #45 179-180 "7" : Int
+        "#]],
+    );
+}
+
+#[test]
+fn struct_copy_cons_too_many_fields() {
+    check(
+        indoc! {"
+            namespace A {
+                struct Pair { First : Int, Second : Int }
+                function Foo() : Pair {
+                    let pair = new Pair { First = 5, Second = 6 };
+                    new Pair { ...pair, First = 7, Second = 8, Third = 9 }
+                }
+            }
+        "},
+        "",
+        &expect![[r#"
+            #19 76-78 "()" : Unit
+            #23 86-211 "{\n        let pair = new Pair { First = 5, Second = 6 };\n        new Pair { ...pair, First = 7, Second = 8, Third = 9 }\n    }" : UDT<"Pair": Item 1>
+            #25 100-104 "pair" : UDT<"Pair": Item 1>
+            #27 107-141 "new Pair { First = 5, Second = 6 }" : UDT<"Pair": Item 1>
+            #32 126-127 "5" : Int
+            #35 138-139 "6" : Int
+            #37 151-205 "new Pair { ...pair, First = 7, Second = 8, Third = 9 }" : UDT<"Pair": Item 1>
+            #40 165-169 "pair" : UDT<"Pair": Item 1>
+            #45 179-180 "7" : Int
+            #48 191-192 "8" : Int
+            #51 202-203 "9" : Int
+            Error(Type(Error(MissingClassCorrectFieldCount("Pair", Span { lo: 151, hi: 205 }))))
+            Error(Type(Error(MissingClassHasField("Pair", "Third", Span { lo: 194, hi: 203 }))))
+        "#]],
+    );
+}
+
+#[test]
+fn struct_cons_udt_not_struct() {
```

Comment:
Would it make sense to have a test that a UDT declared with `newtype` that has the right structure (depth 1, everything named) can be treated as a struct? Something like:
```qsharp
newtype Pair = (First : Int, Second : Int);
function Foo() : Pair { new Pair { First = 1, Second = 2 } }
```

Reply:
done